src/users/schemas/user-settings.schema.ts

```diff
@@ -5,6 +5,43 @@ import { Document } from "mongoose";
 
 export type UserSettingsDocument = UserSettings & Document;
 
+// Define possible filter component types as a union of string literals
+export type FilterComponentType =
+  | "LocationFilterComponent"
+  | "PidFilterComponent"
+  | "PidFilterContainsComponent"
+  | "PidFilterStartsWithComponent"
+  | "GroupFilterComponent"
+  | "TypeFilterComponent"
+  | "KeywordFilterComponent"
+  | "DateRangeFilterComponent"
+  | "TextFilterComponent";
+
+// Define the Filter interface
+export interface FilterConfig {
+  type: FilterComponentType;
+  visible: boolean;
+}
+
+// Define the Condition interface
+export interface ScientificCondition {
+  field: string;
+  value: string;
+  operator: string;
+}
+
+export const kDefaultFilters: FilterConfig[] = [
```

Comment:
k stands for?

Reply:
Generally it would be more appreciated if the naming is consistent with existing naming convention within the application scope.

Reply:
#Sure, what is the current convention?

We don't have a strict naming convention in this project yet, but we should have one documented in the future.
Google also has a code styling standard for [typescript](https://google.github.io/styleguide/tsguide.html) as well, which, generally speaking is like unspoken standard for most of the typescript applications

Reply:
Thanks! I will check that out!